pkgs/by-name/pa/paperlib/package.nix

```
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchurl
+, pkgs
+, appimageTools
+}:
+let
+  pname = "paperlib";
+  version = "3.1.6";
+  src = fetchurl {
+    x86_64-darwin = {
+
+      url = "https://github.com/Future-Scholars/${pname}/releases/download/release-electron-${version}/Paperlib_${version}.dmg";
+      hash = "sha256-d9vEFx59K15PO7DJYJQ2fjiagqa8oJLtoawILDF9IKc=";
+    };
+    x86_64-linux = {
+      url = "https://github.com/Future-Scholars/${pname}/releases/download/release-electron-${version}/Paperlib_${version}.AppImage";
+      hash = "sha256-2xbn9UWlcf37n9jZdZKyyevzsag6SW9YuQH/bYCRmLQ=";
+
+    };
+  }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
+
+  meta = with lib;{
+    homepage = "https://github.com/Future-Scholars/paperlib?tab=readme-ov-file";
+    description = "Open-source academic paper management tool";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ByteSudoer ];
+    platforms = [ "x86_64-darwin" "x86_64-linux" ];
+    mainProgram = "paperlib";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+  };
+in
+if stdenv.isDarwin then
+  stdenv.mkDerivation
+  {
+    inherit pname version src meta;
+
+    sourceRoot = ".";
+
+    nativeBuildInputs = with pkgs; [ undmg ];
```

This should specify `undmg` as an input and let this be just

```suggestion
    nativeBuildInputs = [ undmg ];
```

Otherwise, overrides may not work as expected.